libs/vanilla/src/inputs/check.input.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  defineView,
+  ExtractDefProps,
+  ViewComponentProps,
+  withWrapper,
+} from '@reactlit/core';
+import { DetailedHTMLProps } from 'react';
+
+type BaseCheckInputProps = Omit<
+  DetailedHTMLProps<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    HTMLInputElement
+  >,
+  'value' | 'disabled'
+> & {
+  containerClassName?: string;
+  format?: (value: any) => string | React.ReactNode;
+  keyof?: (value: any) => string | number;
+  valueof?: (value: any) => string | number;
+  options: string[] | Object[];
```

Comment:
Ideally for the object use case you can use a generic T instead of Object. This would diverge, but improve on what Observable does. The ideal typing would be a bit tricky to get right, but we could have generics for the valueof result as well. If you don't feel comfortable doing that it's ok we could improve the typing on a subsequent pass, but if we do get it right you could essentially have the view return typed options or typed option values that are aware of what you provided so you'd get autocomplete on the other end.